R/ggcoef_model.R

```diff
@@ -0,0 +1,739 @@
+#' @param exponentiate if `TRUE` a logarithmic scale will
+#' be used for x-axis
+#' @param point_size size of the points
+#' @param point_stroke thickness of the points
+#' @param point_fill fill colour for the points
+#' @param colour optional variable name to be mapped to
+#' colour aesthetic
+#' @param colour_guide should colour guide be displayed
+#' in the legend?
+#' @param colour_lab label of the colour aesthetic in the legend
+#' @param colour_labels labels argument passed to
+#' [ggplot2::scale_colour_discrete()] and
+#' [ggplot2::discrete_scale()]
+#' @param shape optional variable name to be mapped to the
+#' shape aesthetic
+#' @param shape_values values of the different shapes to use in
+#' [ggplot2::scale_shape_manual()]
+#' @param shape_guide should shape guide be displayed in the legend?
+#' @param shape_lab label of the shape aesthetic in the legend
+#' @param errorbar should error bars be plotted?
+#' @param errorbar_height height of error bars
+#' @param errorbar_coloured should error bars be coloured as the points?
+#' @param stripped_rows should stripped rows be displayed in the background?
+#' @param strips_odd color of the odd rows
+#' @param strips_even color of the even rows
+#' @param vline should a vertical line de drawn at 0 (or 1 if `exponentiate = TRUE`)?
+#' @param vline_colour colour of vertical line
+#' @param dodged should points be dodged (according to the colour aesthetic)?
+#' @param dodged_width width value for [ggplot2::position_dodge()]
+#' @param facet_row variable name to be used for row facets
+#' @param facet_col optional variable name to be used for column facets
+#' @param facet_labeller labeller function to be used for labelling facets;
+#'   if labels are too long, you can use [ggplot2::label_wrap_gen()] (see examples),
+#'   more information in the documentation of [ggplot2::facet_grid()]
+#' @export
+ggcoef_plot <- function (
+  data,
+  x = "estimate",
+  y = "label",
+  exponentiate = FALSE,
+  point_size = 2,
+  point_stroke = 2,
+  point_fill = "white",
+  colour = NULL,
+  colour_guide = TRUE,
+  colour_lab = "",
+  colour_labels = ggplot2::waiver(),
+  shape = "significance",
+  shape_values = c(16, 21),
+  shape_guide = TRUE,
+  shape_lab = "",
+  errorbar = TRUE,
+  errorbar_height = .1,
+  errorbar_coloured = FALSE,
+  stripped_rows = TRUE,
+  strips_odd = "#11111111",
+  strips_even = "#00000000",
+  vline = TRUE,
+  vline_colour = "grey50",
+  dodged = FALSE,
+  dodged_width = .8,
+  facet_row = "var_label",
+  facet_col = NULL,
+  facet_labeller = "label_value"
+){
+  data[[y]] <- forcats::fct_rev(forcats::fct_inorder(data[[y]]))
+  if (!is.null(facet_row))
+    data[[facet_row]] <- forcats::fct_inorder(data[[facet_row]])
+
+
+  if (stripped_rows) {
+    if (!"term" %in% names(data)) {
+      data$term <- data[[y]]
+    }
+    data <- data %>%
+      mutate(.fill = dplyr::if_else(
+        as.integer(forcats::fct_inorder(term)) %% 2L == 1,
```

Comment:
Maybe this will fix this warning?
```
* checking R code for possible problems ... NOTE
ggcoef_plot: no visible binding for global variable ‘term’
Undefined global functions or variables:
  term
```

```suggestion
        as.integer(forcats::fct_inorder(.data$term)) %% 2L == 1,
```